packages/services/schema/src/lib/add-inaccessible-to-unreachable-types.ts

```diff
@@ -0,0 +1,60 @@
+import { DocumentNode, parse, print } from 'graphql';
+import { transformSupergraphToPublicSchema } from '@theguild/federation-composition';
+import { ComposerMethodResult } from './compose';
+import { addDirectiveOnTypes, getReachableTypes } from './reachable-type-filter';
+
+/**
+ * Adds inaccessible directive to unreachable types
+ *
+ * @param resolveName
+ * @param compositionResult
+ * @param supergraphSDL
+ * @returns
+ */
+
+export const addInaccessibleToUnreachableTypes = (
+  resolveName: (identity: string, name: string) => string,
+  compositionResult: ComposerMethodResult,
+  supergraphSDL: DocumentNode,
+): ComposerMethodResult => {
+  const inaccessibleDirectiveName = resolveName(
+    'https://specs.apollo.dev/inaccessible',
+    '@inaccessible',
+  );
+  const federationTypes = new Set([
+    resolveName('https://specs.apollo.dev/join', 'FieldSet'),
+    resolveName('https://specs.apollo.dev/join', 'Graph'),
+    resolveName('https://specs.apollo.dev/link', 'Import'),
+    resolveName('https://specs.apollo.dev/link', 'Purpose'),
+    resolveName('https://specs.apollo.dev/federation', 'Policy'),
+    resolveName('https://specs.apollo.dev/federation', 'Scope'),
+    resolveName('https://specs.apollo.dev/join', 'DirectiveArguments'),
```

Comment:
I'd prefer this not being a hardcoded list, but this is how it was before also.

And I'm not sure how we'd avoid it unless it is guaranteed all of these are namespaced...